packages/now-cli/src/util/input/edit-project-settings.ts

```diff
@@ -4,32 +4,30 @@ import chalk from 'chalk';
 import { Output } from '../output';
 import { Framework } from '@vercel/frameworks';
 import { isSettingValue } from '../is-setting-value';
+import { ProjectSettings } from '../../types';
 
-export interface ProjectSettings {
+export interface PartialSettings {
   buildCommand: string | null;
   outputDirectory: string | null;
   devCommand: string | null;
 }
 
-export interface ProjectSettingsWithFramework extends ProjectSettings {
-  framework: string | null;
-}
-
-const fields: { name: string; value: keyof ProjectSettings }[] = [
+const fields: { name: string; value: keyof PartialSettings }[] = [
   { name: 'Build Command', value: 'buildCommand' },
   { name: 'Output Directory', value: 'outputDirectory' },
   { name: 'Development Command', value: 'devCommand' },
 ];
 
 export default async function editProjectSettings(
   output: Output,
-  projectSettings: ProjectSettings | null,
+  projectSettings: PartialSettings | null,
   framework: Framework | null
-) {
+): Promise<ProjectSettings> {
   // create new settings object, missing values will be filled with `null`
-  const settings: Partial<ProjectSettingsWithFramework> = {
-    ...projectSettings,
-  };
+  const settings: ProjectSettings = Object.assign(
+    { framework: null },
+    projectSettings
+  );
```

Comment:
@AndyBitz I noticed this function was returning `Partial<ProjectSettingsWithFramework>` when it seems to populate all properties. What do you think about my solution here by assigning `framework` right at the beginning so we can return the complete `ProjectSettings` each time?

Reply:
I think this should be fine, since we set it to `null` anyways when it's falsy